web/src/main/java/com/futurewei/alcor/web/entity/dataplane/NeighborInfo.java

```diff
@@ -71,4 +66,22 @@ public String getPortMac() {
     public void setPortMac(String portMac) {
         this.portMac = portMac;
     }
+
+    public NeighborInfo(String hostIp, String hostId, String portId, String portMac) {
+        this.hostIp = hostIp;
+        this.hostId = hostId;
+        this.portId = portId;
+        this.portMac = portMac;
+    }
+
+    @Override
+    public int hashCode() {
+       return super.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        NeighborInfo o=(NeighborInfo)obj;
+        return this.hostId.equals(o.hostId)&&this.hostIp.equals(o.hostIp)&&this.portId.equals(o.portId);
```

Comment:
Do we need to consider port ip and mac when checking equality?

Reply:
i think it depends on our req, e.g if we have same portid, should we consider same port?
same portid would have different portIp or mac?

Reply:
One port id could have multiple port Ips (that is where the fixedIP plays a role) but always with the same mac address. 

This doesn't matter here though as we want to compare all fields of two objects fairly. @haboy52581